manifests/scripts/metrics-dashboard/run.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# -----------------
+
+# Using OpenShift's Prometheus instance requires us to enable user workload monitoring via the openshift-monitoring ConfigMap:
+# https://docs.openshift.com/container-platform/4.11/monitoring/enabling-monitoring-for-user-defined-projects.html)
+# - BUT, the way I'm doing it here is a brittle hack :P
+kubectl apply -f prometheus/enable-user-workload-monitoring.yaml
+
+# Allow Prometheus process to view the openshift-operators/gitops namespaces
+kubectl apply -f prometheus/prometheus-roles-for-openshift-operators.yaml -n openshift-operators
+kubectl apply -f prometheus/prometheus-roles-for-gitops-ns.yaml -n gitops
+
+# -----------------
+echo 
+echo "* Kick off an OLM install of Grafana, and wait for it to complete"
+
+kubectl create ns grafana > /dev/null 2>&1
+
+kubectl apply -f grafana/grafana-operator-group.yaml -n grafana
+kubectl apply -f grafana/grafana-subscription.yaml -n grafana
+
+echo
+echo "* Waiting for Grafana CRDs to exist"
+
+while : ; do
+  kubectl get customresourcedefinition/grafanas.integreatly.org  > /dev/null 2>&1 && break
+  sleep 1s
+done
+
+# -----------------
+
+echo "* Waiting for Grafana route"
+
+while : ; do
+  kubectl get route/grafana-route -n grafana -o yaml  > /dev/null 2>&1 && break
+  sleep 1s
+done
+
+export HOSTNAME=`kubectl get route/grafana-route -n grafana -o yaml  | grep "    host:" | cut -c11-`
+export ADMIN_SECRET_VALUE=`openssl rand -hex 10`
+
+echo
+echo "* Grafana route is: https://$HOSTNAME"
+echo "  Username: user"
+echo "  Password: $ADMIN_SECRET_VALUE"
+echo
+
+TMP_DIR=`mktemp -d`
+
+# Substitute the cluster domain into the Grafana Ingress CR
+cp -f grafana/grafana-cr.yaml $TMP_DIR/grafana-cr-resolved.yaml
+sed -i 's/HOSTNAME/'$HOSTNAME'/g' $TMP_DIR/grafana-cr-resolved.yaml
```

Comment:
On mac, `sed -i` requires a backup suffix, for example `sed -i.bak`, which will also work on linux I believe.